### PR TITLE
Make all public API in dotnet-watch.dll internal and sealed.

### PR DESCRIPTION
**Pull request by tmat**
dotnet-watch.dll has been intended to be used as a tool, not consumed as a library and expose any extensibility points via public APIs.

### CODE REVIEW

File: template_feed/Microsoft.DotNet.Common.ItemTemplates/content/Nuget/.template.config/localize/templatestrings.cs.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿{
+{
```
Comment (vlada-shubina) [inline template_feed/Microsoft.DotNet.Common.ItemTemplates/content/Nuget/.template.config/localize/templatestrings.cs.json:1]: Could you please fix those whitespace changes? Thank you.

**Review by vlada-shubina — DISMISSED** [review]
Please avoid changes in templates as this PR is not related to them.

File: template_feed/Microsoft.DotNet.Common.ItemTemplates/content/Nuget/.template.config/localize/templatestrings.cs.json (hunk 1 above)
Comment (tmat) [reply template_feed/Microsoft.DotNet.Common.ItemTemplates/content/Nuget/.template.config/localize/templatestrings.cs.json]: Good catch. I don't know how these got changed :(

**Review by vlada-shubina — COMMENTED** [review]
Looks good from templating perspective

**Comment by tmat** [comment]
@javiercn PTAL

**Review by javiercn — APPROVED** [review]
Looks correct from what I can tell.

Are these APIs used by VS in some way?

**Comment by tmat** [comment]
> Are these APIs used by VS in some way?

No, VS doesn't use dotnet-watch at all.

**Pull request merged**